llama-index-integrations/llms/llama-index-llms-azure-openai/llama_index/llms/azure_openai/base.py

```diff
@@ -186,6 +186,10 @@ def _get_credential_kwargs(self, **kwargs: Any) -> Dict[str, Any]:
         if self.use_azure_ad:
             self._azure_ad_token = refresh_openai_azuread_token(self._azure_ad_token)
             self.api_key = self._azure_ad_token.token
+        else:
+            import os
+
+            self.api_key = self.api_key or os.getenv("AZURE_OPENAI_API_KEY")
```

Comment:
I don't think `os.getenv` will raise an error if the env var is not specified. But perhaps it should, and moreoever maybe wrap in a try/except block so that we can raise a custom Error & msg?

Reply:
Amazing! Thanks 🙏